demos/common/cpp/models/include/models/internal_model_data.h

```diff
@@ -37,12 +37,20 @@ struct InternalImageModelData : public InternalModelData {
     int inputImgHeight;
 };
 
-struct InternalScaleData : public InternalModelData {
-    InternalScaleData(float scaleX, float scaleY) {
+struct InternalImageMatModelData : public InternalImageModelData {
+    InternalImageMatModelData(int width, int height, const cv::Mat& mat) : InternalImageModelData(width, height), mat(mat) {}
+
+    cv::Mat mat;
+};
+
+struct InternalScaleMatData : public InternalModelData {
+    InternalScaleMatData(float scaleX, float scaleY, const cv::Mat& mat) {
         x = scaleX;
         y = scaleY;
+        this->mat = mat;
```

Comment:
I would suggest switching to using a member initializer list; it would be more consistent with `InternalImageMatModelData` and avoid unnecessary default construction of `mat`.